### DISCUSSION TITLE
How to get extent of the currently visible map?

### DISCUSSION THREAD

**Discussion by urossavelj** [Mapsui/Mapsui#2167] [Q&A] [answered]
So I'm making an app, that will download the tiles with BruTile. I need to pass the extent, resolution etc. of the map. I tried to get the extent via Mapsui.Map.Extent property, but I get everytime the same result: `Extent: BL: (X=-20037508,342789,Y=-20037508,342789)  TR: (X=20037508,342789,Y=20037508,342789)` and it doesn't matter how much I zoom or move the map.

Can anyone give me some insight with this?

**Comment by pauldendulk** [comment, marked as answer]
The Map.Extent contains the extent of the static map definition, so the extent over all data in the layers. The Viewport is the dynamic visible window. You can get the extent with: ```Map.Navigator.Viewport.ToExtent();```

**Reply by urossavelj** [reply]
I think I tried everything besides this. Thanks a lot! It would be helpful if this would be written in the documentation 😄

**Reply by pauldendulk** [reply]
There is room for improvement there. This general page could be improved and more prominent in the docs https://mapsui.com/documentation/mapsui-components.html. Also we could improve our in-code summaries.
